### PR TITLE
[Merged by Bors] - feat: port Topology.Bornology.Constructions

### PR DESCRIPTION
**Pull request by j-loreaux**
porting notes:

1. at one point I had to explicit provide the bornology to `isBounded_univ`. This seems like a regression and should be investigated.

- [x] depends on: #1822

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1822
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1822~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Bornology/Constructions.lean [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+#align bornology.is_bounded.bounded_space_subtype Bornology.IsBounded.boundedSpace_subtype
+
+alias boundedSpace_val_set_iff ↔ _ Bornology.IsBounded.boundedSpace_val
+#align bornology.is_bounded.bounded_space_coe Bornology.IsBounded.boundedSpace_val
+
+instance [BoundedSpace α] {p : α → Prop} : BoundedSpace (Subtype p) :=
+  (IsBounded.all { x | p x }).boundedSpace_subtype
+
+/-!
+### `additive`, `multiplicative`
```
Comment (jcommelin) [inline Mathlib/Topology/Bornology/Constructions.lean:184]:
```suggestion
### `Additive`, `Multiplicative`
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**